src/utils/utils.ts

```diff
@@ -250,62 +233,77 @@ export const isNeedField = ({
     return false;
   }
 
-  const issueType = project.issuetypes.find(
-    ({ id }: JiraIssueType) => id === issueTypeId,
-  );
+  const issueType = project.issuetypes.find(({ id }) => id === issueTypeId);
 
   return issueType?.fields?.[fieldName] !== undefined;
 };
 
 export const jiraIssueToFormValues = (
-  issue: Record<string, any>,
-  usableFields: (Assignee | Attachment)[],
+  issue: IssueBean["fields"],
+  usableFields: FieldMeta[],
 ) => {
-  const values = Object.keys(issue).reduce(
-    (acc, key) => {
-      if (!usableFields.find((e) => e.key === key)) return acc;
+  const values = Object.keys(issue).reduce<JiraIssueSchema>((acc, key) => {
+    const usableField = usableFields.find((e) => e.key === key)
 
-      const usableField = usableFields.find((e) => e.key === key);
+    if (!usableField) {
+      return acc;
+    }
 
-      switch (usableField?.schema.type) {
-        case "user": {
-          acc[key] = {
-            id: issue[key as keyof IssueItem]?.accountId,
-          };
+    switch (usableField?.schema?.type) {
+      case "user": {
+        acc[key] = {
+          id: (issue[key] as UserBean)?.accountId,
+        };
 
-          break;
-        }
+        break;
+      }
 
-        case "date":
-        case "datetime": {
-          if (!issue[key as keyof IssueItem]) break;
-          acc[key] = new Date(issue[key as keyof IssueItem]);
+      case "date":
+      case "datetime": {
+        if (!issue[key]) break;
+        acc[key] = new Date(issue[key] as DateTime);
```

Comment:
Is this a DateTime or a string. If it's from the API where does it get hydrated.

Reply:
`DateTime` is defined in the file `/src/types.ts`

<img width="695" alt="Screenshot 2024-11-21 at 16 23 57" src="https://github.com/user-attachments/assets/fdc591a8-db9f-4aae-a3c2-524771453de3">